pandas/_libs/groupby.pyx

```diff
@@ -1082,27 +1084,38 @@ def group_quantile(
         Array containing the values to apply the function against.
     labels : ndarray[np.intp]
         Array containing the unique group labels.
-    sort_indexer : ndarray[np.intp]
-        Indices describing sort order by values and labels.
     qs : ndarray[float64_t]
         The quantile values to search for.
+    starts : ndarray[int64]
```

Comment:
Do starts and ends need to be passed separately? On initial review I'm expecting that the groups are all monotonic in each array, so does `starts[i]` always equal `ends[i-1] - 1`?

Reply:
lib.generate_slices returns a pair of ndarrays, id have to look at that more closely to comment on the why